File: ObjectProgrammingPython/Rosliny/trawa.py
import random

from Rosliny.roslina import Roslina


class Trawa(Roslina):
    def __init__(self, Swiat, x, y, czyZyje):
        super(Trawa, self).__init__(Swiat, x, y, czyZyje)
        self.sila = 0
        self.inicjatywa = 0
        self.kolor = "lightgreen"
        self.oznaczenie = "t"
        self.czyMozeRozmnazac = 0
        self.wiek = 0

    def akcja(self):
        self.przygotujOrganizm()
        losuj = random.randint(0,100)
        if losuj > 95:
            super().akcja()

    def Wypisz(self):
        return "Trawa" + self.koncowka()

File: ObjectProgrammingJava/projekt po - 2/src/projekt/rosliny/Trawa.java
package projekt.rosliny;

import projekt.Organizm;
import projekt.Wypisanie;
import projekt.swiat.Swiat;
import projekt.zwierzeta.Zolw;

import java.awt.*;
import java.util.Random;
import java.util.Vector;

public class Trawa extends Roslina implements Wypisanie {
    public Trawa(Swiat swiat, int x, int y, boolean czyZyje){
        polozenieX = x;
        polozenieY = y;
        sila = 0;
        inicjatywa = 0;
        kolor = Color.green;
        oznaczenie = 't';
        this.swiat = swiat;
        this.czyZyje = czyZyje;
        this.czyMozeRozmnazac = 0;
        wiek = 0;
    }

    @Override
    public void akcja() {
        przygotujOrganizm();
        Random random = new Random();
        int losuj = random.nextInt(100);
        if(losuj>95){
            super.akcja();
        }
    }

    @Override
    public String Wypisz() {
        return "Trawa" + getKoncowka();
    }
}
